MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioFactory.java

```diff
@@ -42,228 +55,274 @@
+        registerScenario(new HideAndSeekBuiltInScenario());
+        registerScenario(new HoldTheLineBuiltInScenario());
+        registerScenario(new OfficerDualBuiltInScenario());
+        registerScenario(new PirateFreeForAllBuiltInScenario());
+        registerScenario(new PrisonBreakBuiltInScenario());
+        registerScenario(new ProbeBuiltInScenario());
+        registerScenario(new ReconRaidBuiltInScenario());
+        registerScenario(new StandUpBuiltInScenario());
+        registerScenario(new StarLeagueCache1BuiltInScenario());
+        registerScenario(new StarLeagueCache2BuiltInScenario());
+    }
+
+    private AtBScenarioFactory() {
+    }
+
+    public static List<Class<IAtBScenario>> getScenarios(int type) {
+        return scenarioMap.get(type);
+    }
+
+    public static AtBScenario createScenario(Campaign c, Lance lance, int type, boolean attacker, Date date) {
+        List<Class<IAtBScenario>> classList = getScenarios(type);
+        Class<IAtBScenario> selectedClass;
+
+        if ((classList == null) || classList.isEmpty()) {
+            return null;
+        }
+
+        if (classList.size() > 1) {
+            Random randomGenerator = new Random();
+            selectedClass = classList.get(randomGenerator.nextInt(classList.size()));
+        } else {
+            selectedClass = classList.get(0);
+        }
+
+        try {
+            AtBScenario s = (AtBScenario) selectedClass.newInstance();
+            s.initialize(c, lance, attacker, date);
+
+            return s;
+        } catch (InstantiationException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static void registerScenario(IAtBScenario scenario) {
+    final String METHOD_NAME = "registerScenario(IAtBScenario)"; //$NON-NLS-1$
+
+    if (!scenario.getClass().isAnnotationPresent(AtBScenarioEnabled.class)) {
+        MekHQ.getLogger().log(AtBScenarioFactory.class, METHOD_NAME, LogLevel.ERROR,
+                String.format("Unable to register an AtBScenario of class '%s' because is does not have the '%s' annotation.", //$NON-NLS-1$
+        scenario.getClass().getName(), AtBScenarioEnabled.class.getName()));
+    } else {
+        int type = scenario.getScenarioType();
         List<Class<IAtBScenario>> list = scenarioMap.computeIfAbsent(type, k -> new ArrayList<>());
 
         list.add((Class<IAtBScenario>) scenario.getClass());
-
-		return true;
-	}
-
-	/* Iterate through the list of lances and make a battle roll for each,
-	 * then sort them by date before adding them to the campaign.
-	 * Contracts with enemy morale level of invincible have a base attack
-	 * (defender) battle each week. If there is a base attack (attacker)
-	 * battle, that is the only one for the week on that contract.
-	 */
-	public static void createScenariosForNewWeek(Campaign c, boolean allowLancesToBeDuplicated) {
```

Comment:
What was the purpose of removing this parameter?

Reply:
Player confusion. When I described how it works, I got multiple "that's a bug, right" regarding it.